File: crates/ecp-cli/tests/fixtures/incremental_parity/dart/user_service.dart
import 'user.dart';

abstract class UserRepository {
  Future<User?> findById(int id);
  Future<List<User>> findAll();
  Future<User> save(User user);
  Future<bool> delete(int id);
}

class UserService {
  final UserRepository _repository;

  UserService(this._repository);

  Future<User?> getUser(int id) => _repository.findById(id);

  Future<List<User>> listUsers() => _repository.findAll();

  Future<User> createUser(String email, String name) async {
    final user = User(id: DateTime.now().millisecondsSinceEpoch, email: email, name: name);
    return _repository.save(user);
  }

  Future<bool> deleteUser(int id) => _repository.delete(id);
}

File: crates/ecp-cli/tests/fixtures/incremental_parity/swift/UserService.swift
protocol UserRepository {
    func findById(_ id: Int64) -> User?
    func findAll() -> [User]
    func save(_ user: User) -> User
    func delete(_ id: Int64) -> Bool
}

class UserService {
    private let repository: UserRepository

    init(repository: UserRepository) {
        self.repository = repository
    }

    func getUser(id: Int64) -> User? {
        repository.findById(id)
    }

    func listUsers() -> [User] {
        repository.findAll()
    }

    func createUser(email: String, name: String) -> User {
        let user = User(id: Int64(Date().timeIntervalSince1970), email: email, name: name)
        return repository.save(user)
    }

    func deleteUser(id: Int64) -> Bool {
        repository.delete(id)
    }
}
